data-catalog-api/server/src/main/java/org/apache/airavata/datacatalog/api/sharing/SharingManagerImpl.java

```diff
@@ -89,8 +112,29 @@ public void initialize(String tenantId) throws SharingException {
     }
 
     @Override
-    public UserEntity resolveUser(UserInfo userInfo) {
-        return null;
+    public UserEntity resolveUser(UserInfo userInfo) throws SharingException {
+        try (UserManagementClient userManagementClient = custosClientProvider.getUserManagementClient()) {
```

Comment:
This is good but it should first check the database (UserRepository) to see if the user already exists or not. Then it only needs to make the Custos API call if it doesn't exist locally.

Reply:
@machristie I will add that
Please try this 
https://github.com/apache/airavata-custos/tree/develop